tests/byoc/byoc.go

```diff
@@ -0,0 +1,260 @@
+					Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllEnvironmentsInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(fw.UserNamespace)).To(Succeed())
+					Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllGitOpsDeploymentInASpecificNamespace(fw.UserNamespace, 30*time.Second)).To(Succeed())
+					Expect(fw.SandboxController.DeleteUserSignup(fw.UserName)).NotTo(BeFalse())
+				}
+			})
+
+			It("initializes byoc cluster connection", func() {
+				if suite.Byoc.ClusterType == appservice.ConfigurationClusterType_Kubernetes {
+					byocKubeconfig, err = vc.InitializeVCluster(fw.UserNamespace, fw.UserNamespace)
+					Expect(byocKubeconfig).NotTo(BeEmpty(), "failed to initialize vcluster. Kubeconfig not provided")
+					Expect(err).NotTo(HaveOccurred())
+
+				} else if suite.Byoc.ClusterType == appservice.ConfigurationClusterType_OpenShift {
+					byocKubeconfig = utils.GetEnv("BYOC_KUBECONFIG", "")
+					Expect(byocKubeconfig).NotTo(BeEmpty(), "Please provide BYOC_KUBECONFIG env pointing to a valid openshift kubeconfig file")
+				}
+
+				config, err := clientcmd.BuildConfigFromFlags("", byocKubeconfig)
+
+				Expect(err).NotTo(HaveOccurred())
+				byocAPIServerURL = config.Host
+				Expect(byocAPIServerURL).NotTo(BeEmpty())
+
+				ephemeralClusterClient, err = client.NewKubeFromKubeConfigFile(byocKubeconfig)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("creates managed-gitops.redhat.com/managed-environment secret type", func() {
+				kubeConfData, err := os.ReadFile(byocKubeconfig)
+				data := make(map[string][]byte)
+				data["kubeconfig"] = kubeConfData
+				Expect(err).NotTo(HaveOccurred())
+
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      ManagedEnvironmentSecretName,
+						Namespace: fw.UserNamespace,
+					},
+					Type: corev1.SecretType(ManagedEnvironmentType),
+					Data: data,
+				}
+
+				_, err = fw.AsKubeAdmin.CommonController.CreateSecret(fw.UserNamespace, secret)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("creates environment", func() {
+				// Note: It is not possible yet to configure integration service snapshots to deploy to a specific environment.
+				// As an workaround for now: Deletes the development environment and recreate it with kubernetes cluster information
+				Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllEnvironmentsInASpecificNamespace(fw.UserNamespace, 1*time.Minute)).NotTo(HaveOccurred())
+
+				environment, err := fw.AsKubeAdmin.GitOpsController.CreateEphemeralEnvironment(ManagedEnvironmentName, fw.UserNamespace, suite.Byoc.TargetNamespace, byocAPIServerURL, ManagedEnvironmentSecretName, suite.Byoc.ClusterType)
+				Expect(environment.Name).To(Equal(ManagedEnvironmentName))
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("creates namespace in the ephemeral target cluster", func() {
```

Comment:
I feel like this could be moved into the Cluster type class to do this for us when initialized?